backend/apps/ifc_validation/checks/check_bsdd.py

```diff
@@ -1,396 +1,622 @@
+
+
+@functools.lru_cache(maxsize=128)
+def get_classes(uri):
+
+    url = "https://api.bsdd.buildingsmart.org/api/Dictionary/v1/Classes"
+    uri = uri.split("/class")[0]
+    response = requests.get(url, {'Uri':uri})
+    return response.json()['classes'] if response.status_code == 200 else None
+
 
-            model.status_bsdd = 'v'
-            #session.add(bsdd_result)
+@functools.lru_cache(maxsize=128)
+def get_material(uri):
+
+    url = "https://api.bsdd.buildingsmart.org/api/Class/v1"
+    response = requests.get(url, {'Uri':uri, 'ClassType': 'Material'})
+    return response.json() if response.status_code == 200 else None
+
+
+@functools.lru_cache(maxsize=128)
+def get_materials(uri):
+
+    url = "https://api.bsdd.buildingsmart.org/api/Dictionary/v1/Classes"
+    uri = uri.split("/class")[0]
+    response = requests.get(url, {'Uri':uri, 'ClassType': 'Material'})
+    return response.json()['classes'] if response.status_code == 200 else None
+
+
+@functools.lru_cache(maxsize=128)
+def get_properties(uri):
+
+    url = "https://api.bsdd.buildingsmart.org/api/Dictionary/v1/Properties"
+    uri = uri.split("/class")[0]
+    response = requests.get(url, {'Uri':uri})
+    return response.json()['properties'] if response.status_code == 200 else None
 
-        #session.commit()
 
-# WORK IN PROGRESS (port of Flask > Django)
+@functools.lru_cache(maxsize=128)
+def find_property_by_uri(uri):
+
+    url = "https://api.bsdd.buildingsmart.org/api/Property/v4"
+    response = requests.get(url, {'uri': uri})
+    return response.json() if response.status_code == 200 else None
+
+
+def first_available_attr(object, properties):
+
+    for prop in properties:
+
+        #if prop in object.__dir__():
+        if hasattr(object, prop): 
+            return getattr(object, prop)
+        
+    return None
+
+
+#TODO: exploratory code, needs some refactoring
 def perform(file_name, task_id, verbose=False):
+    
+    ifc_file = ifcopenshell.open(file_name)   
+    bsdd_results = {
+        'dictionaries': [],
+        'classes': [],
+        'assignments': [],
+        'properties': [],
+        'materials': [],
+        'messages': []
+    }    
+    
+    ifc_file_classifications = ifc_file.by_type("IfcClassification")
+    ifc_file_classification_references = ifc_file.by_type("IfcClassificationReference")
+    ifc_file_rel_associates_classifications = ifc_file.by_type("IfcRelAssociatesClassification")
+    ifc_file_properties = ifc_file.by_type("IfcProperty")
+    ifc_file_property_sets = ifc_file.by_type("IfcPropertySet")
+    ifc_file_materials = ifc_file.by_type("IfcMaterial")
+    ifc_file_rel_associates_material = ifc_file.by_type("IfcRelAssociatesMaterial")
+
+    # bSDD dictionary (former name: domain)
+    # https://github.com/buildingSMART/bSDD/blob/master/Documentation/bSDD-IFC%20documentation.md#1-bsdd-dictionary
+    if len(ifc_file_classifications):
+
+        for _, ic in enumerate(ifc_file_classifications):
```

Comment:
```suggestion
        for ic in ifc_file_classifications:
```

Why use enumerate if the index goes unreferenced?